src/VisualStudio/Core/Def/CallHierarchy/CallHierarchyCommandHandler.cs

```diff
@@ -39,61 +46,75 @@ internal class CallHierarchyCommandHandler : ICommandHandler<ViewCallHierarchyCo
-                {
-                    return true;
-                }
+                return true;
+            }
+
+            var caretPosition = args.TextView.Caret.Position.BufferPosition.Position;
+
+            // We're showing our own UI, ensure the editor doesn't show anything itself.
+            context.OperationContext.TakeOwnership();
+            var token = _listener.BeginAsyncOperation(nameof(ExecuteCommand));
+            ExecuteCommandAsync(document, caretPosition)
+                .ReportNonFatalErrorAsync()
+                .CompletesAsyncOperation(token);
+
+            return true;
+        }
 
-                var semanticModel = document.GetSemanticModelAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+        private async Task ExecuteCommandAsync(Document document, int caretPosition)
+        {
+            using (var context = _threadOperationExecutor.BeginExecute(
+                ServicesVSResources.Call_Hierarchy, ServicesVSResources.Navigating, allowCancellation: true, showProgress: false))
+            {
+                var cancellationToken = context.UserCancellationToken;
 
-                var caretPosition = args.TextView.Caret.Position.BufferPosition.Position;
-                var symbolUnderCaret = SymbolFinder.FindSymbolAtPositionAsync(
-                    semanticModel, caretPosition, document.Project.Solution.Services, cancellationToken)
-                    .WaitAndGetResult(cancellationToken);
+                var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(true);
+                var symbolUnderCaret = await SymbolFinder.FindSymbolAtPositionAsync(
+                    semanticModel, caretPosition, document.Project.Solution.Services, cancellationToken).ConfigureAwait(true);
 
                 if (symbolUnderCaret != null)
                 {
                     // Map symbols so that Call Hierarchy works from metadata-as-source
                     var mappingService = document.Project.Solution.Services.GetService<ISymbolMappingService>();
-                    var mapping = mappingService.MapSymbolAsync(document, symbolUnderCaret, cancellationToken).WaitAndGetResult(cancellationToken);
+                    var mapping = await mappingService.MapSymbolAsync(document, symbolUnderCaret, cancellationToken).ConfigureAwait(true);
 
                     if (mapping.Symbol != null)
                     {
-                        var node = _provider.CreateItemAsync(mapping.Symbol, mapping.Project, ImmutableArray<Location>.Empty, cancellationToken).WaitAndGetResult(cancellationToken);
+                        var node = await _provider.CreateItemAsync(mapping.Symbol, mapping.Project, ImmutableArray<Location>.Empty, cancellationToken).ConfigureAwait(true);
+
                         if (node != null)
                         {
+                            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
This seems a _bit_ odd since you had ConfigureAwait(trues) everywhere so this should have been on the UI thread in the first place, but I guess there's no harm to it.

Reply:
I believe it was because I added it prior to switching to ConfigureAwait(true)